MyApplication/app/src/main/java/net/serkanbal/myapplication/MainActivity.java

```diff
@@ -0,0 +1,139 @@
+                Log.d("SERKANO", "onReceive: " + value);
+                if (mFirstNumber  == 0 && mSecondNumber == 0) {
+                    mJob1Before.setText("Waiting...");
+                    mSecondNumber = value;
+                    mJob1After.setText(mSecondNumber.toString());
+                } else {
+                    mFirstNumber = mSecondNumber;
+                    mJob1Before.setText(mFirstNumber.toString());
+                    mSecondNumber = value;
+                    mJob1After.setText(mSecondNumber.toString());
+                }
+
+            }
+        };
+
+        mReceiver2 = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                String date = intent.getStringExtra(JobServiceDateTime.MESSAGE2);
+
+                if (mFirstDate.equals("") && mSecondDate.equals("")) {
+                    mJob2Before.setText("Waiting...");
+                    mSecondDate = date;
+                    mJob2After.setText(mSecondDate);
+                } else {
+                    mFirstDate = mSecondDate;
+                    mJob2Before.setText(mFirstDate);
+                    mSecondDate = date;
+                    mJob2After.setText(mSecondDate);
+                }
+
+            }
+        };
+
+        mReceiver3 = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                String token = intent.getStringExtra(JobServiceRandomToken.MESSAGE3);
+
+                if (mFirstToken.equals("") && mSecondToken.equals("")) {
+                    mJob3Before.setText("Waiting...");
+                    mSecondToken = token;
+                    mJob3After.setText(mSecondToken);
+                } else {
+                    mFirstToken = mSecondToken;
+                    mJob3Before.setText(mFirstToken);
+                    mSecondToken = token;
+                    mJob3After.setText(mSecondToken);
+                }
+            }
+        };
+
+        JobInfo periodicJobInfo = new JobInfo.Builder(FIRST_JOB_ID,
+                new ComponentName(this, JobServiceRandomNumber.class))
+                .setPeriodic(5000)
+                .build();
+
+        JobInfo periodicJobInfo2 = new JobInfo.Builder(SECOND_JOB_ID,
+                new ComponentName(this, JobServiceDateTime.class))
+                .setPeriodic(3000)
+                .build();
+
+        JobInfo conditionJobInfo = new JobInfo.Builder(THIRD_JOB_ID,
+                new ComponentName(this, JobServiceRandomToken.class))
+                .setRequiresCharging(true)
+                .setBackoffCriteria(1000, JobInfo.BACKOFF_POLICY_LINEAR)
+                .build();
+
+        JobScheduler jobScheduler = (JobScheduler) getSystemService(JOB_SCHEDULER_SERVICE);
+        jobScheduler.schedule(periodicJobInfo);
+        jobScheduler.schedule(periodicJobInfo2);
+        jobScheduler.schedule(conditionJobInfo);
```

Comment:
You probably saw this coming... I suggest more descriptive names for each job. Something like `numberJob`, `tokenJob`, `dateTimeJob` would be much clearer than these names.